server/meca/export.js

```diff
@@ -9,20 +9,33 @@ const archiveGenerator = require('./file-generators/archive')
 
 const upload = require('./services/upload')
 
-async function generate(manuscript, content, clientIp) {
-  return archiveGenerator({
+async function generate(manuscript, getContent, clientIp) {
+  let content = ''
+  const supportingFiles = {}
+
+  manuscript.files.forEach(file => {
+    if (file.type === 'MANUSCRIPT_SOURCE') {
+      content = getContent(file)
+    } else if (file.type === 'SUPPORTING_FILE') {
+      supportingFiles[file.filename] = getContent(file)
+    }
+  })
+
```

Comment:
May be worth erroring if `content` is null.

Reply:
Think this is the responsibility of the storage class ... pretty certain you'd get a rejected Promise if the `S3` storage fails.

Reply:
I think it's worth doing for `content` though (the manuscript source), as I think the export shouldn't be allowed to happen if it doesn't exist.

An error message like `'Failed to export MECA package. No manuscript source was provided'` makes it clear what the problem is.

Reply:
Sounds good...
@hdrury1 ... should we also fail if a supporting file cannot be put in the zip file? ... or should we carry on regardless?

Reply:
Talking with @Cooryd ... we are thinking the best thing to do is to check against file size (storing it in the db) ... as this would get round the zero size file issue. At the moment my feeling is that we should not do this check as we risk a false error rather than a real one.

/cc @hdrury1 ... do you think this warrants a separate issue?